apps/namadillo/src/atoms/integrations/functions.ts

```diff
@@ -77,6 +78,15 @@ const testnetChains: ChainRegistryEntry[] = [
 
 const mainnetAndTestnetChains = [...mainnetChains, ...testnetChains];
 
+// Terrible hack to inject nam asset in osmosis so we can show them as a token for ibc.
+// We should update the chain-registry and later remove this hack!
+// Please note that this is only a fix for mainnet nam, not housefire
+registry.assets
+  .find((chain) => chain.chain_name === "osmosis")
+  ?.assets.push(mainnetNamAssetOnOsmosis);
+osmosis.assets.assets.push(mainnetNamAssetOnOsmosis);
+// End of the hack
+
```

Comment:
For the future, this PR only introduced this code, the rest is reverting the https://github.com/anoma/namada-interface/pull/2182 

In case you have some ideas to improve this, please check both PRs